src/jdk.sctp/unix/native/libsctp/SctpChannelImpl.c

```diff
@@ -229,7 +229,10 @@ void handleSendFailed
         msg->msg_iovlen = 1;
 
         bufferObj = (*env)->NewDirectByteBuffer(env, addressP, dataLength);
-        CHECK_NULL(bufferObj);
+        if (bufferObj == NULL) {
+            free(addressP);
+            return;
+        }
 
```

Comment:
This looks fine. If NewDirectByteBuffer returns NULL there will be a pending OutOfMemoryError.